### PR TITLE
SLIM-908 Configures HikariDataSource for tests

### PR DESCRIPTION
**Pull request by bvdzwet**
During testing of Bypass Retry, there were database connection
issues. Changing the SingleConnectionDataSource for the Cucumber
tests to the HikariDataSource is an attempt to get rid of these
errors like "PSQLException: This connection has been closed".

### CODE REVIEW

**Comment by bvdzwet** [comment]
retest this please

**Comment by MartijnSipsCGI** [comment]
retest this please

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolDlmsPersistenceConfig.java [hunk 1]
```diff
@@ -27,13 +27,21 @@
...
     @Value("${entitymanager.packages.to.scan.dlms}")
     private String entitymanagerPackagesToScan;
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolDlmsPersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolIec61850PersistenceConfig.java [hunk 2]
```diff
@@ -28,13 +28,21 @@
     public AdapterProtocolIec61850PersistenceConfig() {
     }
 
+    @Value("${db.name.osgp_adapter_protocol_iec61850}")
+    private String databaseName;
+
     @Value("${osgpadapterprotocoliec61850dbs.url}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolIec61850PersistenceConfig.java]: Is this field still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolIec61850PersistenceConfig.java [hunk 3]
```diff
@@ -28,13 +28,21 @@
...
     @Value("${entitymanager.packages.to.scan.iec61850}")
     private String entitymanagerPackagesToScan;
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolIec61850PersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolOslpPersistenceConfig.java [hunk 4]
```diff
@@ -29,13 +29,21 @@
     public AdapterProtocolOslpPersistenceConfig() {
     }
 
+    @Value("${db.name.osgp_adapter_protocol_oslp}")
+    private String databaseName;
+
     @Value("${osgpadapterprotocoloslpdbs.url}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolOslpPersistenceConfig.java]: Is this field still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolOslpPersistenceConfig.java [hunk 5]
```diff
@@ -29,13 +29,21 @@
...
     @Value("${entitymanager.packages.to.scan.oslp}")
     private String entitymanagerPackagesToScan;
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterProtocolOslpPersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsMicrogridsPersistenceConfig.java [hunk 6]
```diff
@@ -28,13 +28,21 @@
     public AdapterWsMicrogridsPersistenceConfig() {
     }
 
+    @Value("${db.name.osgp_adapter_ws_microgrids}")
+    private String databaseName;
+
     @Value("${osgpadapterwsmicrogridsdbs.url}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsMicrogridsPersistenceConfig.java]: Is this fields still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsMicrogridsPersistenceConfig.java [hunk 7]
```diff
@@ -28,13 +28,21 @@
...
     @Value("${entitymanager.packages.to.scan.ws.microgrids}")
     private String entitymanagerPackagesToScan;
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsMicrogridsPersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsSmartMeteringPersistenceConfig.java [hunk 8]
```diff
@@ -26,6 +26,9 @@
     basePackageClasses = { MeterResponseDataRepository.class })
 public class AdapterWsSmartMeteringPersistenceConfig extends ApplicationPersistenceConfiguration {
 
+    @Value("${db.name.osgp_adapter_ws_smartmetering}")
+    private String databaseName;
+
     @Value("${osgpadapterwssmartmeteringdbs.url}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsSmartMeteringPersistenceConfig.java]: Is this field still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsSmartMeteringPersistenceConfig.java [hunk 9]
```diff
@@ -36,6 +39,11 @@ public AdapterWsSmartMeteringPersistenceConfig() {
     }
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/AdapterWsSmartMeteringPersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/ApplicationPersistenceConfiguration.java [hunk 10]
```diff
@@ -62,10 +76,12 @@
     @Value("${hibernate.show_sql}")
     protected String hibernateShowSql;
 
+    protected abstract String getDatabaseName();
+
     protected abstract String getDatabaseUrl();
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/ApplicationPersistenceConfiguration.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/CorePersistenceConfig.java [hunk 11]
```diff
@@ -27,6 +27,9 @@
         DeviceRepository.class, RtuDeviceRepository.class })
 public class CorePersistenceConfig extends ApplicationPersistenceConfiguration {
 
+    @Value("${db.name.osgp_core}")
+    private String databaseName;
+
     @Value("${osgpcoredbs.url}")
     private String databaseUrl;
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/CorePersistenceConfig.java]: Is this field still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/CorePersistenceConfig.java [hunk 12]
```diff
@@ -63,6 +66,11 @@ public LocalContainerEntityManagerFactoryBean entityMgrCore(@Qualifier("dsCore")
     }
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/CorePersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/LoggingPersistenceConfig.java [hunk 13]
```diff
@@ -25,6 +25,9 @@
 @EnableJpaRepositories(entityManagerFactoryRef = "entityMgrLogging", transactionManagerRef = "txMgrLogging", basePackageClasses = { DeviceLogItemRepository.class })
 public class LoggingPersistenceConfig extends ApplicationPersistenceConfiguration {
 
+    @Value("${db.name.osgp_logging}")
+    private String databaseName;
+
     @Value("${osgploggingdbs.url}")
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/LoggingPersistenceConfig.java]: Is this field still needed?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/LoggingPersistenceConfig.java [hunk 14]
```diff
@@ -35,6 +38,11 @@ public LoggingPersistenceConfig() {
     }
 
     @Override
+    protected String getDatabaseName() {
+        return this.databaseName;
+    }
+
+    @Override
     protected String getDatabaseUrl() {
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/config/LoggingPersistenceConfig.java]: Is this property still needed?

File: cucumber-tests-platform/src/test/resources/cucumber-platform.properties [hunk 15]
```diff
@@ -9,17 +9,30 @@ certificate.basepath=../certificates
...
+
+db.name.osgp_core=osgp_core
+db.name.osgp_logging=osgp_logging
+db.name.osgp_adapter_ws_smartmetering=osgp_adapter_ws_smartmetering
+db.name.osgp_adapter_protocol_dlms=osgp_adapter_protocol_dlms
+db.name.osgp_adapter_protocol_oslp=osgp_adapter_protocol_oslp
+db.name.osgp_adapter_ws_microgrids=osgp_adapter_ws_microgrids
+db.name.osgp_adapter_protocol_iec61850=osgp_adapter_protocol_iec61850
+
+osgpcoredbs.url=${db.protocol}${db.hostname}:${db.port}/${db.name.osgp_core}
```
Comment (smvdheijden) [inline cucumber-tests-platform/src/test/resources/cucumber-platform.properties]: Are the URL properties still needed?

**Pull request merged**